### PR TITLE
update text input and text area components

### PR DESCRIPTION
**Pull request by jennyvb143**
Completed the following tasks per updated requirements:
Text Area and Text Input: Update text color to be darker (currently it's very light)
Text Area and Text Input: Set required to true by default
Text Area and Text Input: Append "(Optional)" to label text when required is false
Text Input: Remove the following types from available options until they are styled properly (may never happen): button time image date checkbox color datetime-local file hidden month radio range reset submit week
Text Area: Remove styles for 100% width when `cols` has a value

Clubhouse Story: https://app.clubhouse.io/ebth/story/2020/create-input-component-text-based-text-area

### CODE REVIEW

**Review by tonytrucco-CS — APPROVED** [review]
The changes look good!

File: src/molecules/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -13,29 +13,7 @@ import {
...
-    | 'range'
-    | 'reset'
-    | 'search'
-    | 'submit'
-    | 'tel'
-    | 'text'
-    | 'time'
-    | 'url'
-    | 'week';
+  type?: 'email' | 'number' | 'password' | 'search' | 'tel' | 'text' | 'url';
```
Comment (mosherc) [inline src/molecules/TextInput/TextInput.tsx:16]: @tonytrucco-CS what's wrong with date, datetime-local, month, time, and week? Those are all text based as well.

**Review by mosherc — APPROVED** [review]
Code looks good. Just one question for Tony if you don't mind waiting until he answers.

File: src/molecules/TextInput/TextInput.tsx (hunk 1 above)
Comment (tonytrucco-CS) [reply src/molecules/TextInput/TextInput.tsx:16]: @mosherc I don't have an explicit issue with them, but my concern is with their specific styling needs (if any) that may not have been addressed or considered this early on in the component's life. If they are visually bullet-proof then they can be added back into the rotation. I felt a more limited set that we have full confidence in made more sense from a "guard rails" perspective for any devs that might want to use the component.

Comment (jennyvb143) [reply src/molecules/TextInput/TextInput.tsx:16]: I would say that, as-is, they're not visually bulletproof. The date/time-centric input types add an interactive calendar icon to the UI that is in direct conflict with our tooltip icon. So we would need to address how to handle that in order to include these types.

Comment (mosherc) [reply src/molecules/TextInput/TextInput.tsx:16]: @jennyvb143 oh I don't know why I was unable to see that, probs cuz I was on Safari. Sounds good then

**Pull request merged**